src/router_pynode.c

```diff
@@ -452,11 +456,17 @@ qd_error_t qd_router_python_setup(qd_router_t *router)
     pySetMobileSeq   = PyObject_GetAttrString(pyRouter, "setMobileSeq"); QD_ERROR_PY_RET();
     pySetMyMobileSeq = PyObject_GetAttrString(pyRouter, "setMyMobileSeq"); QD_ERROR_PY_RET();
     pyLinkLost       = PyObject_GetAttrString(pyRouter, "linkLost"); QD_ERROR_PY_RET();
+//    Py_DECREF(adapterInstance);  // TODO: why not this? get python exceptions if I try
```

Comment:
I understand now. `PyTuple_SetItem` does not incref, it transfers ownership from local context into the tuple. Meaning that doing decref here (or for the other tuple items before) is a mistake.

Reply:
http://www.cse.psu.edu/~gxt29/papers/refcount.pdf: "When programmers use thoseAPI functions, they can often be confusedby their effects on refcounts and makemistakes." I can see why that might be the case.